lib/parsers/composer-parser.ts

```diff
@@ -33,65 +34,125 @@ export class ComposerParser {
 
   public static buildDependencies(
     composerJsonObj: ComposerJsonFile,
-    composerLockObjPackages: LockFilePackage[],
+    composerLockObj: ComposerLockFile,
     depObj: ComposerJsonFile | LockFilePackage,
-    depRecursiveArray: string[],
-    systemVersions: SystemPackages,
+    systemPackages: SystemPackages,
     includeDev = false,
-    packageReferencesCount: PackageRefCount = {}): DepTree | {} {
-    const requires: ComposerDependencies | undefined = _.get(depObj, 'require', undefined);
-    if (!requires) {
-      return {};
+    isDevTree = false,
+    depRecursiveArray: string[] = [],
+    packageRefCount: PackageRefCount = {},
+  ): DepTreeDependencies {
+
+    const result: DepTreeDependencies = {};
+
+    // find depObj properties
+    const depName = _.get(depObj, 'name');
```

Comment:
what  if there is no `depName`? the default will be null here I guess we do not want to proceed.

Reply:
Since this is recursive does it mean we can end up with a top level or transitive dep with no name? `no_project_name` only makes sense when a top level package has no name. Am I missing something perhaps?

Reply:
I hope it name would be missing in a dependency, it would be invalid Composer file. But not sure about it.

Reply:
It would be invalid composer file, but I think because nested dependency names are referenced by key like so:

```
"require": {
    "php": "^7.1"
},
"require-dev": {
    "doctrine/coding-standard": "^6.0",
    "ext-pdo": "*",
    "ext-phar": "*",
    "phpbench/phpbench": "^0.13",
    "phpstan/phpstan-phpunit": "^0.11",
    "phpstan/phpstan-shim": "^0.11",
    "phpunit/phpunit": "^7.0"
},
```

If one of these didn't have a name, it wouldn't exist and we wouldn't even process.
If the name was missing from lock file, would wouldn't find it and so break out of the recursion at that point.

I could add test for invalid lock file (package with no name)?

Reply:
I've added test when lockfile contains package with no name, here we assert that the name and version are taken from composer.json instead